### PR TITLE
Adds anosmia quirk

### PR DESCRIPTION
**Pull request by NeonNik2245**
## About The Pull Request

Adds quirk anosmia (0 points), that makes character don't feel any smells.

## How This Contributes To The Nova Sector Roleplay Experience
Now characters can have quirk and don't feel any smells of this world. For bad or good...

## Proof of Testing

image

<details>
<summary>Formulations</summary>

Now it's:
Desc: For some reasons you can't smell the smells 
Gain text: You can't tell if you feel something's smell. 
Lose text: Your nose seemed to clear up, and you began to feel all the smells of this world! 
Medical records: Patient can't fell any smells. 
</details>
<details>
<summary>Screenshots/Videos</summary>
 
image
image

</details>

## Changelog

:cl:
add: Adds anosmia quirk
/:cl:

### CODE REVIEW

**Comment by NeonNik2245** [comment]
My English is not very good, so I will gladly accept the wording changes, especially "smell the smells"

**Comment by NeonNik2245** [comment]
Aaand I don't know, why gateway_test failed on /datum/unit_test/spy_bounty 
> Error: No bounties were generated on first refresh for difficulty Hard
FAILURE #1: No bounties were generated on first refresh for difficulty Hard at code/modules/unit_tests/spies.dm:41

File: code/datums/quirks/neutral_quirks/anosmia.dm [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/datum/quirk/item_quirk/anosmia
+	name = "Anosmia"
+	desc = "For some reasons you can't smell the smells."
+	icon = FA_ICON_HEAD_SIDE_COUGH_SLASH
+	value = 0
+	mob_trait = TRAIT_ANOSMIA
+	gain_text = span_notice("You can't tell if you feel something's smell.")
+	lose_text = span_danger("Your nose seemed to clear up, and you began to feel all the smells of this world!")
+	medical_record_text = "Patient can't fell any smells."
```
Comment (SomeRandomOwl) [inline code/datums/quirks/neutral_quirks/anosmia.dm:9]:
```suggestion
/datum/quirk/item_quirk/anosmia
	name = "Anosmia"
	desc = "For some reasons you can't smell the smells."
	icon = FA_ICON_HEAD_SIDE_COUGH_SLASH
	value = 0
	mob_trait = TRAIT_ANOSMIA
	gain_text = span_notice("You suddenly notice the lack of smells, Your nose doesn't seem to work..")
	lose_text = span_danger("Your nose seemed to clear up, and you begin to smell things around you!")
	medical_record_text = "Patient lost the the sensation of smell."
```

File: modular_nova/modules/pollution/code/pollution.dm [hunk 2]
```diff
@@ -38,6 +38,8 @@
 
 /// When a user smells this pollution
 /datum/pollution/proc/smell_act(mob/living/sniffer)
+	if(HAS_TRAIT(sniffer, TRAIT_ANOSMIA)) //Anosmia holders can not feel the smells, so return. NOVA change
+		return
 	var/list/singleton_cache = SSpollution.singletons
```
Comment (SomeRandomOwl) [inline modular_nova/modules/pollution/code/pollution.dm:43]:
```suggestion
	if(HAS_TRAIT(sniffer, TRAIT_ANOSMIA)) //Anosmia holders can not smell things.
		return
	var/list/singleton_cache = SSpollution.singletons
```

**Review by SomeRandomOwl — CHANGES_REQUESTED** [review]
Grammar pass, this wording makes a bit more sense

File: code/datums/quirks/neutral_quirks/anosmia.dm (hunk 1 above)
Comment (OrionTheFox) [reply code/datums/quirks/neutral_quirks/anosmia.dm:9]: double period in gain_text 😭

**Comment by NeonNik2245** [comment]
> Transcendent Olfaction is also going to need to be affected by this. And Pheromone Receptors probably? Make those changes modular if it is possible.
> 
> Nothing else really comes to mind...but I may be missing some.
> 
> There are also various descriptions around that mention smell but you can ignore those as it would be too many edits for something cosmetic.

I also found forge, oven and deep frier smells, that's the most strange or hard to bypass, another is miasma in air, that gain you mood event, so it also has check for now

**Comment by NeonNik2245** [comment]
Can I add "Do not merge" by myself? Just wonder

**Comment by Iajret** [comment]
> Can I add "Do not merge" by myself? Just wonder

as i said already, convert PR to draft while you change every smell to account for your quirk.

**Comment by NeonNik2245** [comment]
Make this quirk on upstream, so after that will change this pr

File: code/datums/mutations/olfaction.dm [hunk 3]
```diff
@@ -40,6 +40,10 @@
 		to_chat(owner, span_warning("You have no nose!"))
 		return FALSE
 
+	if(HAS_TRAIT(living_cast_on, TRAIT_ANOSMIA)) //Anosmia quirk holder cannot fell any smell
+		to_chat(owner, span_warning("You can't feel any smells!"))
```
Comment (RatFromTheJungle) [inline code/datums/mutations/olfaction.dm:44]:
```suggestion
		to_chat(owner, span_warning("You can't smell anything anymore!"))
```

File: code/datums/mutations/olfaction.dm [hunk 4]
```diff
@@ -40,6 +40,10 @@
 		to_chat(owner, span_warning("You have no nose!"))
 		return FALSE
 
+	if(HAS_TRAIT(living_cast_on, TRAIT_ANOSMIA)) //Anosmia quirk holder cannot fell any smell
```
Comment (RatFromTheJungle) [inline code/datums/mutations/olfaction.dm:43]:
```suggestion
	if(HAS_TRAIT(living_cast_on, TRAIT_ANOSMIA)) // Anosmia quirk holders can't smell anything
```

File: code/datums/quirks/neutral_quirks/anosmia.dm [hunk 5]
```diff
@@ -0,0 +1,15 @@
+/datum/quirk/item_quirk/anosmia
+	name = "Anosmia"
+	desc = "For some reasons you can't smell the smells."
```
Comment (RatFromTheJungle) [inline code/datums/quirks/neutral_quirks/anosmia.dm:3]:
```suggestion
	desc = "For some reason, you can't smell anything.
```

File: code/datums/quirks/neutral_quirks/anosmia.dm [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+	icon = FA_ICON_HEAD_SIDE_COUGH_SLASH
+	value = 0
+	mob_trait = TRAIT_ANOSMIA
+	gain_text = span_notice("You suddenly notice the lack of smells, Your nose doesn't seem to work.")
```
Comment (RatFromTheJungle) [inline code/datums/quirks/neutral_quirks/anosmia.dm:7]:
```suggestion
	gain_text = span_notice("You find yourself unable to smell anything!
```

File: code/datums/quirks/neutral_quirks/anosmia.dm [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+	lose_text = span_danger("Your nose seemed to clear up, and you begin to smell things around you!")
```
Comment (RatFromTheJungle) [inline code/datums/quirks/neutral_quirks/anosmia.dm:8]:
```suggestion
	lose_text = span_danger("Suddenly, you can smell again!!")
```

File: code/modules/antagonists/changeling/powers/pheromone_receptors.dm [hunk 8]
```diff
@@ -23,6 +23,9 @@
 /datum/action/changeling/pheromone_receptors/sting_action(mob/living/carbon/user)
 	..()
 	var/datum/antagonist/changeling/changeling = IS_CHANGELING(user)
+	if(HAS_TRAIT(user, TRAIT_ANOSMIA)) //Anosmia quirk holder cannot feel any smell
+		to_chat(user, span_warning("You can't feel any smells!"))
```
Comment (RatFromTheJungle) [inline code/modules/antagonists/changeling/powers/pheromone_receptors.dm:27]:
```suggestion
		to_chat(user, span_warning("You can't smell anything!"))
```

File: code/modules/antagonists/changeling/powers/pheromone_receptors.dm [hunk 9]
```diff
@@ -23,6 +23,9 @@
 /datum/action/changeling/pheromone_receptors/sting_action(mob/living/carbon/user)
 	..()
 	var/datum/antagonist/changeling/changeling = IS_CHANGELING(user)
+	if(HAS_TRAIT(user, TRAIT_ANOSMIA)) //Anosmia quirk holder cannot feel any smell
```
Comment (RatFromTheJungle) [inline code/modules/antagonists/changeling/powers/pheromone_receptors.dm:26]:
```suggestion
	if(HAS_TRAIT(user, TRAIT_ANOSMIA)) //Anosmia quirk holders can't smell anything
```

File: code/modules/surgery/organs/internal/lungs/_lungs.dm [hunk 10]
```diff
@@ -466,6 +466,8 @@
 			miasma_disease.name = "Unknown"
 			breather.AirborneContractDisease(miasma_disease, TRUE)
 	// Miasma side effects
+	if (HAS_TRAIT(breather, TRAIT_ANOSMIA)) //Anosmia quirk holder cannot feel miasma, but can take disease
```
Comment (RatFromTheJungle) [inline code/modules/surgery/organs/internal/lungs/_lungs.dm:469]:
```suggestion
	if (HAS_TRAIT(breather, TRAIT_ANOSMIA)) //Anosmia quirk holder cannot smell miasma, but can get diseases from it.
```

File: modular_nova/modules/mold/code/mold_disease.dm [hunk 11]
```diff
@@ -28,7 +28,8 @@
 				affected_mob.emote("twitch")
 				to_chat(affected_mob, span_danger("You twitch involuntarily. That's not right."))
 			if(SPT_PROB(2, seconds_per_tick))
-				to_chat(affected_mob, span_danger("You sniff, smelling green slime. Does green have a smell?"))
+				if(!HAS_TRAIT(affected_mob, TRAIT_ANOSMIA)) //Anosmia holders can't feel the smell, even if green smells strange
```
Comment (RatFromTheJungle) [inline modular_nova/modules/mold/code/mold_disease.dm:31]:
```suggestion
				if(!HAS_TRAIT(affected_mob, TRAIT_ANOSMIA)) //Anosmia holders can't smell this.
```

File: code/datums/diseases/advance/symptoms/fire.dm [hunk 12]
```diff
@@ -71,7 +71,10 @@
 	if(prob(33.33))
 		living_mob.show_message(span_hear("You hear a crackling noise."), type = MSG_AUDIBLE)
 	else
-		to_chat(living_mob, span_warning("[pick("You feel hot.", "You smell smoke.")]"))
+		if(HAS_TRAIT(living_mob, TRAIT_ANOSMIA)) //Anosmia quirk holder cannot fell any smell
```
Comment (RatFromTheJungle) [inline code/datums/diseases/advance/symptoms/fire.dm:74]:
```suggestion
		if(HAS_TRAIT(living_mob, TRAIT_ANOSMIA)) //Anosmia quirk holder can't smell anything.
```

**Review by RatFromTheJungle — COMMENTED** [review]
grammar checks

**Comment by annikaRU** [comment]
I'd gladly have "can't smell the smells" kinda funny

**Comment by NeonNik2245** [comment]
> See #1890 , I think this can be closed now?

Yes, thank you, I'll make PR to complete #1890 with perfumes and little more

**Pull request closed without merging**